src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
-    node_->get_parameter(name + ".robot_radius", robot_radius_);
-    // Define  tool radius (radius) parameter
-    float tool_radius_default = 0.5f;
-    declare_parameter_if_not_declared(node_, name + ".tool_radius", rclcpp::ParameterValue(tool_radius_default));
-    node_->get_parameter(name + ".tool_radius", tool_radius_);
-    initialized_ = true;
   }
-}
 
-std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool> > const& grid, std::list<gridNode_t>& init,
-                                        std::vector<std::vector<bool> >& visited)
-{
-  int dx, dy, dx_prev, x2, y2, nRows = grid.size(), nCols = grid[0].size();
-  // Spiral filling of the open space
-  // Copy incoming list to 'end'
-  std::list<gridNode_t> pathNodes(init);
-  // Create iterator for gridNode_t list and let it point to the last element of end
-  std::list<gridNode_t>::iterator it = --(pathNodes.end());
-  if (pathNodes.size() > 1)  // if list is length 1, keep iterator at end
-    it--;                    // Let iterator point to second to last element
-
-  gridNode_t prev = *(it);
-  bool done = false;
-  while (!done)
+  SpiralSTC::~SpiralSTC()
+  {
+    RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Destroying plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+  }
+
+  void SpiralSTC::configure(
+      const rclcpp_lifecycle::LifecycleNode::WeakPtr &parent,
+      std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+      std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
   {
-    if (it != pathNodes.begin())
+    if (!initialized_)
     {
-      // turn ccw
-      dx = pathNodes.back().pos.x - prev.pos.x;
-      dy = pathNodes.back().pos.y - prev.pos.y;
-      dx_prev = dx;
-      dx = -dy;
-      dy = dx_prev;
+      // Get node from parent
+      node_ = parent.lock();
+      name_ = name;
+
+      // Currently this plugin does not use the costmap, instead request a map from a server
+      // This will change in the future
+      costmap_ = costmap_ros->getCostmap();
+      global_frame_ = costmap_ros->getGlobalFrameID();
+
+      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"),
+                  "Configuring plugin %s of type NavfnPlanner", name_.c_str());
+
+      // Create a publisher to visualize the plan
+      plan_pub_ = node_->create_publisher<nav_msgs::msg::Path>("plan", 1);
+      // Try to request the cpp-grid from the cpp_grid map_server
+      cpp_grid_client_ = node_->create_client<nav_msgs::srv::GetMap>("/map_server/map");
+
+      // Define  robot radius (radius) parameter
+      float robot_radius_default = 0.5f;
+      declare_parameter_if_not_declared(node_, name_ + ".robot_radius", rclcpp::ParameterValue(robot_radius_default));
+      node_->get_parameter(name_ + ".robot_radius", robot_radius_);
+      // Define  tool radius (radius) parameter
+      float tool_radius_default = 0.5f;
```

Comment:
ROS2 parameters cannot be floats, they are always doubles. So I'd use that as type to avoid conversions.